### PR TITLE
Add unit tests for/refactor stubbed_command

### PR DESCRIPTION
**Pull request by mdurban**

### CODE REVIEW

**Comment by mdurban** [comment]
One test from the osd-platform stuff failed when testing against this branch. Will look more into it tomorrow

File: lib/rspec/shell/expectations/call_log.rb [hunk 1]
```diff
@@ -17,8 +17,7 @@ def called_with_args?(*args)
 
         def stdin_for_args(*args)
           call = find_call(*args)
-          return call['stdin'] if call
-          nil
+          call['stdin'] unless call.nil?
```
Comment (mdurban) [inline lib/rspec/shell/expectations/call_log.rb]: Apparently this broke the test I mentioned in the comments

File: lib/rspec/shell/expectations/call_log.rb [hunk 2]
```diff
@@ -38,7 +37,7 @@ def get_argument_count(*expected_argument_series, sub_command_series: [], positi
         end
 
         def called_with_no_args?
-          load_call_log_list.first["args"].nil?
+          load_call_log_list.first["args"].to_a.empty?
```
Comment (mdurban) [inline lib/rspec/shell/expectations/call_log.rb]: I'm not 100% sure on this change, but I think we may have a bug the way it was before

File: lib/rspec/shell/expectations/call_log.rb [hunk 3]
```diff
@@ -38,7 +37,8 @@ def get_argument_count(*expected_argument_series, sub_command_series: [], positi
         end
 
         def called_with_no_args?
-          load_call_log_list.first["args"].nil?
+          call_log_list = load_call_log_list
+          !call_log_list.empty? and call_log_list.first["args"].nil?
```
Comment (mdurban) [inline lib/rspec/shell/expectations/call_log.rb]: I think we may have had a bug here. If there was no call log file found, the rescue in the load_call_log_list method returned an [ ], which we would then be calling nil.first... on

File: lib/rspec/shell/expectations/stubbed_command.rb [hunk 4]
```diff
@@ -23,8 +23,7 @@ def with_args(*args)
         end
 
         def called?
-          return false unless @call_log.exist?
-          @call_log.called_with_args?(*@args)
+          @call_log.exist? and @call_log.called_with_args?(*@args)
```
Comment (mdurban) [inline lib/rspec/shell/expectations/stubbed_command.rb]: @call_log&.called_with_args?(*@args) in ruby 3

**Comment by bennyhat** [comment]
I give this my approval, btw.

**Comment by mdurban** [comment]
Ok, I'm going to make that one change we talked with Topher about Friday evening (the return nil to make more explicit) and then we can merge it

**Pull request merged**